openml-h2o/src/main/java/com/feedzai/openml/h2o/algos/AbstractH2OAlgoUtils.java

```diff
@@ -0,0 +1,176 @@
+    protected abstract T commonParams(Frame trainingFrame, DatasetSchema datasetSchema);
+
+    /**
+     * Abstract method that creates and returns the Model that this class refers to.
+     *
+     * @param h2oParams concrete implementation of {@link ModelParametersSchemaV3 algorithm params}.
+     * @return concrete implementation of {@link ModelBuilder H20 Model Builder} created from the h2oParams.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public abstract M getModel(T h2oParams);
+
+    /**
+     * Validates the h20 model parameters by building a model and validating it.
+     *
+     * @param paramsToValidate H20 parameters to validate.
+     * @param randomSeed       The source of randomness.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public void validateParams(Map<String, String> paramsToValidate, final long randomSeed){
+        final T baseParams = getEmptyParams();
+        final T updatedParams = parseSpecificParams(baseParams, paramsToValidate, randomSeed);
+        M modelBuilder = getModel(updatedParams);
+        validateModel(modelBuilder);
+    }
+
+    /**
+     * Trains the H20 model.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param schema        The dataset schema.
+     * @return Job resulting from he model training.
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    public Job train(final Frame trainingFrame,
+                     final Map<String, String> params,
+                     final long randomSeed,
+                     final DatasetSchema schema){
+        M model = getModel(trainingFrame, params, randomSeed, schema);
+        return model.trainModel();
+
+    }
+
+    /**
+     * Gets the ModelBuilder for a training scenario.
+     *
+     * @param trainingFrame The dataset to be used.
+     * @param params        The raw training params.
+     * @param randomSeed    The source of randomness.
+     * @param schema        The dataset schema.
+     * @return concrete {@link ModelBuilder H20 Model Builder} implementation
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    private M getModel(final Frame trainingFrame,
+                       final Map<String, String> params,
+                       final long randomSeed,
+                       final DatasetSchema schema) {
+        final T parsedParams = parseParams(trainingFrame, params, randomSeed, schema);
+        return getModel(parsedParams);
+    }
+
+    /**
+     * Method that validates a {@link ModelBuilder H20 Model Builder}
+     *
+     * @param model the ModelBuilder to validate
+     * @throws IllegalArgumentException if the model is invalid
+     *
+     * @since @@@feedzai.next.release@@@
+     */
+    private void validateModel(final M model) {
+        if (model.error_count() > 0) {
+            //FIXME: create a custom exception to throw ?
```

Comment:
you can use `com.feedzai.openml.provider.exception.ModelLoadingException` for this case

Reply:
I mean it should be a `com.feedzai.openml.provider.descriptor.fieldtype.ParamValidationError`

this because it is called only in `com.feedzai.openml.h2o.H2OModelCreator#validateForFit` and is constructing a list of exceptions, so this error should be returned and now throw. to then append to the list of ParamValidationError